.github/workflows/bundle-size.yml

```diff
@@ -0,0 +1,170 @@
+name: Measure bundle size
+
+on:
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  statuses: write
+
+jobs:
+  measure:
+    runs-on: ubuntu-latest
+    steps:
+      # ------- Setup -----------------------------------------------------
+      - uses: actions/github-script@v6
+        name: Create a pending status on the commit
+        with:
+          script: |
+            await github.rest.repos.createCommitStatus({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              sha: context.payload.pull_request.head.sha,
+              context: "Bundle size",
+              state: 'pending',
+              target_url: `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`
+            })
+
+      - name: Checkout the PR branch
+        uses: actions/checkout@v3
+        with:
+          path: 'components'
+
+      - name: Setup the bundle measurement project
+        run: |
+          cp -r components/.github/workflows/bundle-size bundle-size
+          cd bundle-size
+          npm i --force
+
+      # ------- Baseline size measurement ---------------------------------
+
+      - name: Restore the cache for the baseline size measurement
+        uses: actions/cache/restore@v3
+        id: restore-cache-baseline
+        with:
+          path: bundle-size/output-baseline.txt
+          key: bundlesize-${{ hashFiles('bundle-size/build.js','bundle-size/index.html','bundle-size/main.jsx','bundle-size/package-lock.json','bundle-size/package.json') }}
+
+      - name: Measure the baseline size of the Vite project
+        if: steps.restore-cache-baseline.outputs.cache-hit != 'true'
+        run: |
+          cd bundle-size
+          npm run build
+          mv output.txt output-baseline.txt
+          cat output-baseline.txt
+
+      - name: Cache the baseline size
+        uses: actions/cache/save@v3
+        if: steps.restore-cache-baseline.outputs.cache-hit != 'true'
+        with:
+          path: bundle-size/output-baseline.txt
+          key: ${{ steps.restore-cache-baseline.outputs.cache-primary-key }}
+
+      # ------- Base branch size measurement ------------------------------
```

Comment:
Instead of building it from scratch, can we have a separate workflow storing the baseline size when building the mainline and then fetch this info with github API?


For example, board components fetch reference screenshots this way:

https://github.com/cloudscape-design/board-components/blob/2c590d64f2f688a057ef20bfb6fa51854777956d/.github/workflows/visual-regression.yml#L26-L33